src/jit/emitxarch.cpp

```diff
@@ -13017,13 +13017,16 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
         {
             emitGetInsAmdCns(id, &cnsVal);
             code = insCodeRM(ins);
+            code = AddVexPrefixIfNeeded(ins, code, size);
```

Comment:
I don't believe this is the right fix....

The `Is4ByteSSE4OrAVXInstruction()` code paths tend to have slightly different handling for this (within `emitOutput*`, generally).

I think you need to update `emitOutputAM` to get this working properly... In this case, it looks like its failing because `emitOutputAM` only calls `insEncode3456` for `DstDstSrc` or `DstSrcSrc` AVX instructions for which `INS_blendvps` is not included (https://github.com/fiigii/coreclr/blob/54c2e6f8e14b8ba349880e7e83eec5f97a7c4561/src/jit/emitxarch.cpp#L9379).

Reply:
Will change, thanks!

Reply:
It would help if this entire mess was cleaned up to be more consistent, where possible 😄

Reply:
I think we have a bug tracking it already though

Reply:
Fixed.